### PR TITLE
chore(http): migrate to typescript

### PR DESCRIPTION
**Pull request by romainseb**
**What is the problem this PR is trying to solve?**

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/http/src/generators.ts [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
- * @param  {object} options options to deal with cmf automatically
  * @example
  * import { http } from '@talend/http/generators';
  * import { call } from 'redux-saga/effects'
  * yield call(http.post, '/foo', {foo: 42});
  */
-export function* httpPost(url, payload, config, options) {
-	return yield* httpFetch(url, config, HTTP_METHODS.POST, payload, options);
+export function* httpPost(url: string, payload: any, config?: TalendRequestInit) {
+	// @ts-ignore
```
Comment (gjou-tlnd) [inline packages/http/src/generators.ts:17]: why do you need this ?

**Review by gjou-tlnd — CHANGES_REQUESTED** [review]
please don't use ts-ignore in new code..

File: packages/http/src/generators.ts (hunk 1 above)
Comment (romainseb) [reply packages/http/src/generators.ts:17]: I have noooo idea of what type to put in a generator that return a yield* for an async function.
I try to dig a bit 🙄

File: packages/http/src/http.common.test.ts [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+		expect(encodePayload({}, test)).toEqual('FooBar');
+	});
+
+	it('should not json stringify the payload if content-type is not application/json', () => {
+		const headers = {
+			'Content-Type': 'plain/text',
+		};
+		const test = { abc: 'def' };
+
+		// eslint-disable-next-line quotes
```
Comment (tlnd-mhuchet) [inline packages/http/src/http.common.test.ts:166]:
```suggestion
```

File: packages/http/src/http.common.test.ts [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+		expect(encodePayload(headers, test)).toEqual({ abc: 'def' });
+	});
+
+	it('should not json stringify the payload if it is a FormData instance', () => {
+		const headers = {
+			'Content-Type': 'application/json',
+		};
+
+		// eslint-disable-next-line quotes
```
Comment (tlnd-mhuchet) [inline packages/http/src/http.common.test.ts:175]:
```suggestion
```

File: packages/http/src/http.common.test.ts [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+		};
+		const test = { abc: 'def' };
+
+		// eslint-disable-next-line quotes
+		expect(encodePayload(headers, test)).toEqual('{"abc":"def"}');
+	});
+	it('should return the payload as it is if it is a string', () => {
+		const test = 'FooBar';
+
+		// eslint-disable-next-line quotes
```
Comment (tlnd-mhuchet) [inline packages/http/src/http.common.test.ts:156]:
```suggestion
```

File: packages/http/src/generators.ts (hunk 1 above)
Comment (gjou-tlnd) [reply packages/http/src/generators.ts:17]: Actually, I don't really care as it's in generators files, I'll withdraw my change request

**Review by gjou-tlnd — APPROVED** [review]

File: packages/http/package.json [hunk 5]
```diff
@@ -17,10 +17,10 @@
   "main": "lib/index.js",
   "mainSrc": "src/index.ts",
   "scripts": {
+    "build:lib": "talend-scripts build:ts:lib",
```
Comment (jmfrancois) [inline packages/http/package.json:20]: you should add the types entry here to point to lib/index.d.ts
ui-scripts is supposed to warn you about that 🤔

Comment (romainseb) [reply packages/http/package.json:20]: CleanShot 2022-02-02 at 09 07 51@2x

Nothing looks to be warned.
I will add the entry in the package.json

**Review by tlnd-mhuchet — APPROVED** [review]

File: packages/http/package.json (hunk 5 above)
Comment (jmfrancois) [reply packages/http/package.json:21]: it has been added with @talend/scripts-core@11.5.0 and we are at 11.4.0

**Pull request merged**